hetmech/hetmat/archive.py

```diff
@@ -39,23 +41,76 @@ def create_archive_by_globs(
         source_paths -= set(root_directory.glob(glob))
     source_paths = [path.relative_to(root_directory) for path in source_paths]
     source_paths.extend(map(pathlib.Path, include_paths))
-    create_archive(destination_path, root_directory, source_paths, **kwargs)
+    return create_archive(destination_path, root_directory, source_paths, **kwargs)
 
 
-def create_archive(destination_path, root_directory, source_paths, zip_mode='x', compression=zipfile.ZIP_LZMA):
+def create_archive(
+        destination_path, root_directory, source_paths,
+        zip_mode='x', compression=zipfile.ZIP_LZMA, split_size=None):
     """
     Create a zip archive of the source paths at the destination path.
     source_paths as paths relative to the hetmat root directory.
+    split_size is the max zip file size in GB before spliting the zip into
+    multiple parts. The default of split_size=None suppresses splitting.
+    Returns the paths of files this function has written as a list. The first
+    path is a TSV with information on each archived file.
     """
     root_directory = pathlib.Path(root_directory)
-    destination_path = pathlib.Path(destination_path)
     assert zip_mode in {'w', 'x', 'a'}
-    zip_file = zipfile.ZipFile(destination_path, mode=zip_mode, compression=compression)
     source_paths = sorted(set(map(str, source_paths)))
+    destination_path = pathlib.Path(destination_path)
+    if split_size is None:
+        zip_path = destination_path
+    else:
+        zip_path_formatter = f'{destination_path.stem}-{{:04d}}{destination_path.suffix}'.format
+        split_num = 0
+        zip_path = destination_path.with_name(zip_path_formatter(split_num))
+    zip_paths = [zip_path]
+    zip_file = zipfile.ZipFile(zip_path, mode=zip_mode, compression=compression)
     for source_path in source_paths:
         source_fs_path = root_directory.joinpath(source_path)
+        if split_size is not None:
+            fs_size_gb = source_fs_path.stat().st_size / 1e9
+            zip_size_gb = zip_path.stat().st_size / 1e9
+            if zip_file.namelist() and zip_size_gb + fs_size_gb > split_size:
+                zip_file.close()
+                split_num += 1
+                zip_path = destination_path.with_name(zip_path_formatter(split_num))
+                zip_paths.append(zip_path)
+                zip_file = zipfile.ZipFile(zip_path, mode=zip_mode, compression=compression)
         zip_file.write(source_fs_path, source_path)
     zip_file.close()
+    info_df = get_archive_info_df(zip_paths)
+    info_path = destination_path.with_name(destination_path.name + '-info.tsv')
+    info_df.to_csv(info_path, sep='\t', index=False)
+    return [info_path] + zip_paths
+
+
+def get_archive_info_df(zip_paths):
+    """
+    Return member file info for a list of zip archives.
+    """
+    fields = [
+        'filename',
+        'file_size',
+        'compress_type',
```

Comment:
Would be nice to convert `compress_type` to a description rather than int as per https://github.com/python/cpython/blob/6020d98beaf6b9945beb7560f9a1df361e0ba4dd/Lib/zipfile.py#L620-L638